hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheMergeOperationFactory.java

```diff
@@ -83,14 +83,14 @@ public void readData(ObjectDataInput in) throws IOException {
         partitions = in.readIntArray();
         //noinspection unchecked
         mergeEntries = new List[partitions.length];
-        for (int i = 0; i < partitions.length; i++) {
-            List<SplitBrainMergeEntryView<Data, Data>> list = new LinkedList<SplitBrainMergeEntryView<Data, Data>>();
+        for (int partitionIndex = 0; partitionIndex < partitions.length; partitionIndex++) {
```

Comment:
Just to avoid confusion: This is not the `partitionId`, since it's not all partitions, just the ones which are owner by the member this operation is sent to. So this is a `partitionIndex` to iterate over the `partitionId`s in `partitions`.